src/main/java/org/micks/DiscGolfApplication/events/DiscGolfEventService.java

```diff
@@ -126,4 +126,52 @@ public void editEvents(String eventId, DiscGolfEventDTO discGolfEventDTO) {
             throw new RuntimeException("Error while editing event with id: " + eventId, e);
         }
     }
+
+    public void deleteEvent(String eventId) {
+        log.info("Deleting event with id: {}", eventId);
+        try (Connection connection = dbConnection.connect()) {
+            connection.setAutoCommit(false);
+
+            validateEventExists(connection, eventId);
+            deleteUserEvents(connection, eventId);
+            deleteEventRecord(connection, eventId);
+
+            connection.commit();
+        } catch (SQLException e) {
+            log.error("Error while deleting event with id: {}", eventId, e);
+            throw new RuntimeException("Error while deleting event with id: " + eventId, e);
+        }
+    }
+
+    private void validateEventExists(Connection connection, String eventId) throws SQLException {
+        try (PreparedStatement checkStatement = connection.prepareStatement(
+                "SELECT COUNT(*) FROM Events WHERE id = ?")) {
+            checkStatement.setString(1, eventId);
+            ResultSet rs = checkStatement.executeQuery();
+            if (rs.next() && rs.getInt(1) == 0) {
+                throw new IllegalArgumentException("Event with id " + eventId + " not found");
+            }
+        }
+    }
+
+    private void deleteUserEvents(Connection connection, String eventId) throws SQLException {
+        try (PreparedStatement stmt = connection.prepareStatement(
+                "DELETE FROM user_event WHERE event_id = ?")) {
```

Comment:
I think we discussed to mark event as deleted by changing status to DELETED, rather that doing pure delete.